rackcity/api/objects/rack_range.py

```diff
@@ -0,0 +1,44 @@
+from rest_framework import serializers
+
+
+class RackRangeSerializer(serializers.Serializer):
+    letter_start = serializers.CharField(required=True, max_length=1)
+    letter_end = serializers.CharField(required=False, max_length=1)
+    num_start = serializers.IntegerField(required=True)
+    num_end = serializers.IntegerField(required=False)
+
+    def validate(self, data):
+        if ('num_end' in data) and (
+            data['num_end'] < data['num_start']
+        ):
+            message = "invalid rack number range from " + \
+                str(data['num_start']) + \
+                " to " + str(data['num_end'])
+            raise serializers.ValidationError(message)
+
+        if ('letter_end' in data) and (
+            data['letter_end'] < data['letter_start']
+        ):
+            message = "invalid rack row range from " + \
+                data['letter_start'] + " to " + data['letter_end']
+            raise serializers.ValidationError(message)
+
+        return data
+
+    def get_row_range(self, data):
```

Comment:
Django will not allow these methods to be called BEFORE an `.is_valid()` check is called (how Serializer class is implemented), so guaranteed that at least the start fields will exist on data